src/RequestListener.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+namespace Opstalent\CrudBundle;
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\HttpKernel\Event\GetResponseEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Class RequestListener
+ * @author Szymon Kunowski <szymon.kunowski@gmail.com>
+ * @package Opstalent\CrudBundle
+ */
+class RequestListener implements EventSubscriberInterface
+{
+    /**
+     * @var FormConfigResolver
+     */
+    protected $resolver;
+
+    /**
+     * @var FormFactory
+     */
+    protected $factory;
+
+    /**
+     * RequestListener constructor.
+     * @param FormConfigResolver $resolver
+     * @param FormFactory $factory
+     */
+    public function __construct(FormConfigResolver $resolver, FormFactory $factory)
+    {
+        $this->resolver = $resolver;
+        $this->factory = $factory;
+    }
+
+    /**
+     * @return array
+     */
+    public static function getSubscribedEvents()
+    {
+        return [
+            KernelEvents::REQUEST => [
+                ['handleForm', -255],
+            ],
+        ];
+    }
+
+    /**
+     * @param GetResponseEvent $event
+     * @return FormInterface|void
+     */
+    public function handleForm(GetResponseEvent $event)
+    {
+        if (!$event->isMasterRequest()) {
+            return;
+        }
+
+        $request = $event->getRequest();
+        /**
+         * @var CrudHandlingInterface $crud
+         */
+        if (!$crud = $this->resolveCrudRequest($request)) {
+            return;
+        }
+
+        $formModel = $this->resolver->resolve($crud->getAction(), $crud->getClassName());
+        $form = $this->factory->createForm($formModel);
+        $form->handleRequest($request);
+
+        return $form;
+    }
+
+    /**
+     * @param Request $request
+     * @return mixed|null|CrudHandlingInterface
+     */
+    protected function resolveCrudRequest(Request $request)
```

Comment:
return typehint: `: ?CrudHandlingInterface`

Reply:
mismatching in doc: `@return mixed|null|CrudHandlingInterface`. Also I don't get it at all - could you explain how it should works? What is 'crud' attribute? 
Better could be `isCrudRequest()` imho. Also instead returning nulls or stuff like `FormInterface|void` maybe there is better solution e.g. throwing exception

Reply:
I've described it below. typehint will be valid solution so PHP will throw `TypeError` if returned value is not matching typehint

Reply:
attributes->get return mixed|null and we search for CrudHandlingInterface thats why this look like this but now it will be null|CrudHandlingInterface yes ?